tests/system_tests/test_artifacts/test_wandb_artifacts_full.py

```diff
@@ -693,7 +693,7 @@ def test_new_draft(wandb_init):
     # We would use public properties, but they're only available on non-draft artifacts.
     assert draft._entity == parent.entity
     assert draft._project == parent.project
-    assert draft._source_name == art.name
+    assert draft._source_name == art.name.split(":")[0]
```

Comment:
how did it work before?

Reply:
I do not know. @jlzhao27 -- you made this change in your old PR as well, do you know why concurrency changes the behavior here?

Reply:
i can't remember why i made the change or if it is safe anymore. I think the rationale is that `_source_name` should only contain the name and not the version/alias which is after the colon. So the original test was asserting it is `my-sample-portfolio:latest` when the correct behavior should be `my-sample-portfolio` even in prod. I'd double check with the artfact team and on prod to see what `draft._source_name` should be, the test could just be wrong?

Reply:
OK, I figured it out. The `art.name` used to be `test-artifact`, but because I added `artifact.wait()` in `run.link_artifact()`, the name gets updated to `test-artifact:v0`. There's a comment on `self._name` in `artifact.py::Artifact.init` saying "includes version after saving".

TBH, although adding `artifact.wait()` in `link_artifact()` doesn't change much from the user's perspective---we block until we finish uploading and linking either way---it's not the right way to do this. We should instead synchronize log_artifact and link_artifact operations in the internal process. I'll work on this as a follow-up.